src/utils/components/SelectTypeahead/SelectTypeahead.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { FC, ReactNode, Ref, useRef, useState } from 'react';
+
+import {
+  MenuToggleElement,
+  Select,
+  SelectList,
+  SelectOption,
+  SelectOptionProps,
+} from '@patternfly/react-core';
+import { useNetworkingTranslation } from '@utils/hooks/useNetworkingTranslation';
+import { isEmpty } from '@utils/utils';
+
+import Toggle from './Toggle';
+import { filterOptions } from './utils';
+
+type SelectTypeaheadProps = {
+  canCreate?: boolean;
+  id: string;
+  newOptionComponent?: (inputValue: string) => ReactNode;
+  options: SelectOptionProps[];
+  placeholder: string;
+  selected: string;
+  setSelected: (newSelection: null | string) => void;
+};
+
+const SelectTypeahead: FC<SelectTypeaheadProps> = ({
+  canCreate = true,
+  id,
+  newOptionComponent,
+  options,
+  placeholder,
+  selected,
+  setSelected,
+}) => {
+  const { t } = useNetworkingTranslation();
+  const [isOpen, setIsOpen] = useState(false);
+  const [inputValue, setInputValue] = useState<string>(selected || '');
+  const [focusedItemIndex, setFocusedItemIndex] = useState<null | number>(null);
+  const textInputRef = useRef<HTMLInputElement>();
+
+  const selectOptions = inputValue ? filterOptions(options, inputValue) : options;
+
+  const onSelect = (value: string) => {
+    if (value) {
+      setSelected(selected === value ? null : value);
+
+      if (selected === value) setInputValue('');
+
+      setIsOpen(false);
+    }
+
+    textInputRef.current?.focus();
+  };
+
+  return (
+    <Select
+      id={id}
+      isOpen={isOpen}
+      onOpenChange={() => setIsOpen(false)}
+      onSelect={(ev, selection) => onSelect(selection as string)}
+      selected={selected}
+      toggle={(toggleRef: Ref<MenuToggleElement>) => (
+        <Toggle
+          focusedItemIndex={focusedItemIndex}
+          inputValue={inputValue}
+          isOpen={isOpen}
+          onSelect={onSelect}
+          placeholder={placeholder}
+          selected={selected}
+          selectOptions={selectOptions}
+          setFocusedItemIndex={setFocusedItemIndex}
+          setInputValue={setInputValue}
+          setIsOpen={setIsOpen}
+          setSelected={setSelected}
+          textInputRef={textInputRef}
+          toggleRef={toggleRef}
+        />
+      )}
+    >
+      <SelectList id="select-typeahead-listbox">
+        {selectOptions.map((option, index) => (
+          <SelectOption
+            className={option.className}
+            id={`select-typeahead-${option.value.replace(' ', '-')}`}
+            isFocused={focusedItemIndex === index}
+            key={option.value || option.children}
+            {...option}
+          >
+            {option.children || option.value}
+          </SelectOption>
+        ))}
+      </SelectList>
+      {!isEmpty(inputValue) && canCreate && (
```

Comment:
once we entered a single character we will get the new option component, PF suggest to use it when we search and we don't have anymore options, is that the desired behavior in our case?

Reply:
Yes. Not in the case of projects but this is a general component copied by kubevirt and can be useful in the future